File: tests/end-to-end/sam_nodejs_lambda_arr_transform/csharp/Stack.cs
using Amazon.CDK;
using Amazon.CDK.AWS.SAM;
using Constructs;
using System.Collections.Generic;

namespace SAMNodeJSLambdaArr
{
    public class SAMNodeJSLambdaArrProps : StackProps
    {
    }

    public class SAMNodeJSLambdaArr : Stack
    {
        public SAMNodeJSLambdaArr(Construct scope, string id, SAMNodeJSLambdaArrProps props = null) : base(scope, id, props)
        {
            // Transforms
            AddTransform("AWS::Serverless-2016-10-31");

            // Resources
            var myFunction = new CfnFunction(this, "MyFunction", new CfnFunctionProps
            {
                Runtime = "nodejs20.x",
                Handler = "index.handler",
                InlineCode = @"exports.handler = async (event) => {
                  console.log(event);
                }
                ",
            });
        }
    }
}

File: tests/end-to-end/sam_nodejs_lambda_arr_transform/typescript/stack.ts
import * as cdk from 'aws-cdk-lib';
import * as sam from 'aws-cdk-lib/aws-sam';

export interface SAMNodeJSLambdaArrProps extends cdk.StackProps {
}

export class SAMNodeJSLambdaArr extends cdk.Stack {
  public constructor(scope: cdk.App, id: string, props: SAMNodeJSLambdaArrProps = {}) {
    super(scope, id, props);

    // Transforms
    this.addTransform('AWS::Serverless-2016-10-31');

    // Resources
    const myFunction = new sam.CfnFunction(this, 'MyFunction', {
      runtime: 'nodejs20.x',
      handler: 'index.handler',
      inlineCode: 'exports.handler = async (event) => {\n  console.log(event);\n}\n',
    });
  }
}
